docs/adr/2022-04-26__alerting.md

```diff
@@ -0,0 +1,295 @@
+    (udf_js.jackknife_percentile_ci(percentile, measured_values.values).low <= udf_js.jackknife_percentile_ci(percentile, ref.values).low AND udf_js.jackknife_percentile_ci(percentile, ref.values).low <= udf_js.jackknife_percentile_ci(percentile, measured_values.values).high) OR
+    (udf_js.jackknife_percentile_ci(percentile, measured_values.values).high <= udf_js.jackknife_percentile_ci(percentile, ref.values).high AND udf_js.jackknife_percentile_ci(percentile, ref.values).low <= udf_js.jackknife_percentile_ci(percentile, measured_values.values).high) OR
+    (udf_js.jackknife_percentile_ci(percentile, ref.values).low <= udf_js.jackknife_percentile_ci(percentile, measured_values.values).low AND udf_js.jackknife_percentile_ci(percentile, measured_values.values).low <= udf_js.jackknife_percentile_ci(percentile, ref.values).high)) AS ci_overlap,
+    percentile
+  FROM measured_values, 
+    UNNEST([50]) AS percentile  -- percentiles to check
+  JOIN measured_values AS ref
+  ON measured_values.submission_date = ref.submission_date AND
+    measured_values.branch != ref.branch AND
+    measured_values.probe = ref.probe
+  WHERE ref.branch = "tab-unloading-disabled" -- reference branch
+),
+
+ewma AS (
+  -- compute EWMA  
+  SELECT
+    submission_date,
+    probe,
+    branch,
+    udf_js.jackknife_percentile_ci(50, values) as ci,
+    ewma(ARRAY_AGG(udf_js.jackknife_percentile_ci(50, values).percentile) OVER (PARTITION BY branch, probe ORDER BY submission_date DESC), 0.49) AS ewma_percentile
+  FROM measured_values
+)
+
+-- checks for thresholds
+SELECT
+  submission_date,
+  probe,
+  branch,
+  50 AS percentile,
+  "Value below threshold" AS message
+FROM measured_values
+WHERE 
+  udf_js.jackknife_percentile_ci(50, values).high <= 22 AND probe = "gc_max_pause_2"
+-- UNION ALL
+-- ...
+
+-- checks for differences in CI
+UNION ALL
+SELECT
+  submission_date,
+  probe,
+  branch,
+  percentile,
+  "Significant difference between branches" AS message
+FROM ci_overlaps
+WHERE 
+  probe IN ('gc_ms', 'gc_max_pause_2_content', 'js_pageload_delazification_ms')  -- list of probes check is done for
+  AND ci_overlap = FALSE
+
+-- checks for significant changes
+UNION ALL
+SELECT
+    submission_date,
+    probe,
+    branch,
+    50 AS percentile,
+    "Significant difference to historical data" AS message
+FROM ewma 
+WHERE probe IN ("gc_ms")
+  AND (ewma_percentile.ewma - ewma_percentile.stdev > ci.high OR ewma_percentile.ewma + ewma_percentile.stdev < ci.low)
+```
+
+The query currently does not account for external configurations, but can be easily changed to incorporate those.
+
+
+
+## Resources
+
+* https://assets.dynatrace.com/content/dam/en/wp/Anomaly-Detection-for-Monitoring-Ruxit.pdf
```

Comment:
Overall I think this is really great! I think you've picked a good approach and using the statistical process control/quality control framework is the right way to go about it, especially for safety metrics where we don't expect any difference between the groups. 

When I'm thinking about using OpMon for rollouts, there are times when we _would expect_ a difference, such as if we saw one in a prior experiment. Can we can create views in OpMon that replicate the methodology used in MozAnalysis so that the views align? As an example, experiments often measure impacts as aggregated over 4 weeks, however so far we've proposed a framework for making comparisons on a daily basis. If we see a small impact as measured over 4 weeks, we probably won't see a statistically significant difference on any given day, which may cause confusion for our users. Also in this situation where we expect differences, we might want to alert if we see no differences! 

Again, looks great so far!

Reply:
Currently, opmon does only support daily aggregations. It should be possible to extend it to also support weekly, 28 days etc, but that's something that needs to be implemented.

There is also the problem when comparing with jetstream that the statistics jestream applies are more complicated and different than opmon's. But if it becomes a more and more common request that users want to be able to compare or use jetstream metrics for operational monitoring, we might want to invest some time in making both compatible with each other.